src/operator/tensor/broadcast_reduce-inl.h

```diff
@@ -0,0 +1,206 @@
+/*!
+ *  Copyright (c) 2015-2017 by Contributors
+ * \file broadcast_reduce_kernel.h
+ * \brief Function defintion of elementwise unary operators
+ */
+#ifndef MXNET_OPERATOR_TENSOR_BROADCAST_REDUCE_INL_H_
+#define MXNET_OPERATOR_TENSOR_BROADCAST_REDUCE_INL_H_
+
+#include <mxnet/operator_util.h>
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <utility>
+#include "../mshadow_op.h"
+#include "../elemwise_op_common.h"
+#include "./elemwise_binary_op.h"
+#include "../operator_common.h"
+
+namespace mxnet {
+namespace op {
+namespace broadcast {
+using namespace mshadow;
+
+const int MAX_DIM = 5;
+using CShape = Shape<MAX_DIM>;
+
+template<int ndim>
+MSHADOW_XINLINE Shape<ndim> calc_stride(const Shape<ndim>& shape) {
```

Comment:
what's stride for? mxnet tensor doesn't have stride. Does this mean something different?